decentraland/unity-explorer
# Fix: Road assets are not being unloaded when jumping to a world (Memory leak) (#1992)

https://github.com/decentraland/unity-explorer/issues/1992#issuecomment-2363695874

## What does this PR change?

Now all road assets are destroyed when navigating to a different world. Pools are pre-warmed again when going back to Genesis.
No more memory accumulation, no more overlapping road asset duplicates.

## How to test the changes?

1. Launch the explorer in Unity Editor
2. Move around for a while. You will see how road assets are either enabled or created (see ROAD_ASSET_POOL object in the scene).
3. Jump to another world, like 'olavra'.
4. All road assets are gone.
5. Jump to Genesis again.
6. See that new road assets have been created.
7. Select some of those road assets and change its position to check visually that there aren't duplicated assets underneath.
